### ISSUE TITLE
Support for React Navigation V6

### ISSUE THREAD

**Issue by bronzeson** [DataDog/dd-sdk-reactnative#149]
**Is your feature request related to a problem? Please describe.**
Support React Navigation V6

**Describe the solution you'd like**
I would like this SDK to support React Navigation V6.

**Describe alternatives you've considered**
We have been using V6 for a while and we would not like to downgrade.

**Additional context**
When I try to install mobile-react-navigation package, I get this error:

```npm install @datadog/mobile-react-navigation
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: delphi-rn-client@1.0.0
npm ERR! Found: @react-navigation/native@6.0.8
npm ERR! node_modules/@react-navigation/native
npm ERR!   @react-navigation/native@"^6.0.6" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @react-navigation/native@"^5.6.0" from @datadog/mobile-react-navigation@1.0.0-rc4
npm ERR! node_modules/@datadog/mobile-react-navigation
npm ERR!   @datadog/mobile-react-navigation@"*" from the root project
```

**Comment by 0xnm** [comment]
Hello @bronzeson! Thanks for reporting the issue. We will take a look on it on our end and do the necessary changes.

**Comment by jacquesdev** [comment]
Hello @0xnm 

Not quite sure if it's related to this issue, but wanted to confirm - is there a current minimum supported of react-nagivation?

Specifically, does it currently support react-navigation 4 and 5?

**Comment by 0xnm** [comment]
Hello @jacquesdev! SDK supports `React Navigation` v5, but cannot say about v4 - we didn't test the compatibility with that version.

**Comment by ecexplorer** [comment]
@0xnm This is a major blocker for our team. What's the holdup on v6 support?

**Comment by 0xnm** [comment]
Hello @ecexplorer and @bronzeson. I've tested the compatibility of our package with React Navigation v6 and it is working find for the latest version (`6.0.10`).

We still need to update the code to have a better compatibility with v6, because we are using some API (for tracking nested navigators correctly) which was public in v5, but became private in v6, so it can be removed at any point, but in general you can start using React Navigation v6 with our SDK right now already (and we recommend to pin the version of React Navigation for now).

Let us know if you face any issues.

**Comment by jacobbodkin** [comment]
@0xnm I can confirm that `1.0.0-rc5` works with React Navigation V6 but am having issues with nested navigators. Specifically looking to track launch times for different entry points in the app via the RUM dashboard and only the lowest level navigator is available in `resolveNavigationStateChangeListener` (aka no `state` to loop through).

Can you confirm that is one of the issues cause by the `public` to `private` change?

**Comment by louiszawadzki** [comment]
HI @jacobbodkin!
We're working on adding support for React Navigation v6 here: https://github.com/DataDog/dd-sdk-reactnative/pull/187 and it should be released pretty soon :) 

To make sure we're also fixing your issue, could you specify what you mean by tracking "launch times for different entry points in the app": do you mean opening the app using a deeplink to go to a specific nested navigator? Or navigating to a nested navigator from another screen?

**Comment by louiszawadzki** [comment]
Hi everyone, 

Support for React Navigation v6 has been released in version 1.0.0-rc7!
Make sure you update both the `@datadog/mobile-react-native` and `@datadog/mobile-react-navigation` packages. In most cases, no update in your code should be needed.
